src/mono/mono/utils/write-protect.h

```diff
@@ -0,0 +1,18 @@
+/**
+ * \file
+ */
+
+#ifndef __MONO_WRITE_PROTECT_H__
+#define __MONO_WRITE_PROTECT_H__
+
+#include <mono/utils/mono-publib.h>
+
+#if defined (HOST_MACCAT) && defined (__aarch64__)
```

Comment:
I thought this was supposed to not only apply to catalyst?

Reply:
Agreed; wouldn't you want `if defined(HOST_DARWIN_SIMULATOR)`?

Reply:
oops. you're right